src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -153,6 +153,13 @@ def is_sku_available(cmd, sku_info, zone):
     return is_available
 
 
+def is_os_disk(DiskCreateOption, disk_create_option, os_type):
+    if disk_create_option == DiskCreateOption.from_image or disk_create_option in \
+            [DiskCreateOption.import_enum, DiskCreateOption.upload, DiskCreateOption.empty] and os_type:
```

Comment:
From this comment https://github.com/Azure/azure-cli/issues/23614#issuecomment-1243476098, we can know 

> Disk will be OS disk if:
    - If the `CreateOption` is set to **FromImage**
    - If the `CreateOption` is Import, Upload or Empty and `--os-type` parameter is provided.
```suggestion
    if disk_create_option == DiskCreateOption.from_image or (disk_create_option in \
            [DiskCreateOption.import_enum, DiskCreateOption.upload, DiskCreateOption.empty] and os_type):
```

Reply:
@AjKundnani May I ask that `import_secure` and `upload_prepared_secure` also belongs to OS disk, right?

Reply:
It is recommended not to write too complex logic in `if` judgment condition. Please split it into multiple meaningful variables and then judge those variables in `if` statement

Reply:
> @AjKundnani May I ask that `import_secure` and `upload_prepared_secure` also belongs to OS disk, right?

I referenced the mapping relationship of SDK and service field in https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/compute/azure-mgmt-compute/azure/mgmt/compute/v2022_03_02/models/_compute_management_client_enums.py#L46-L77. I thought maybe only DiskCreateOption.import_enum is related to Import, since the import_secure is mapped to ImportSecure and upload_prepared_secure is mapped to UploadPreparedSecure. I am not sure if it is correct understanding.

Reply:
> @AjKundnani May I ask that `import_secure` and `upload_prepared_secure` also belongs to OS disk, right?

@zhoxing-ms - At present, import_secure and upload_prepared_secure are used for OS disk, right. But in future it could scale to data disk scenario as well.

Reply:
> So if we can check for parameter --os-type along with same, that'll be OS disk scenario.

@AjKundnani Do you think the following logic is in line with your expectations?
```suggestion
    if disk_create_option == DiskCreateOption.from_image or (disk_create_option in \
            [DiskCreateOption.import_enum, DiskCreateOption.import_secure, DiskCreateOption.upload_prepared_secure, DiskCreateOption.upload, DiskCreateOption.empty] and os_type):
```